File: python/src/strings/longest_substring.py
class Solution:
    def lengthOfLongestSubstring(self, inputString: str) -> int:
        maxLength = 0
        startIndex = 0
        seen = {}
        for index, letter in enumerate(inputString):
            if letter in seen:
                maxLength = max(maxLength, index - startIndex)
                startIndex = max(startIndex, seen[letter] + 1)
            seen[letter] = index
        return max(maxLength, len(inputString) - startIndex)

File: java/src/main/java/com/blind75/strings/LongestSubString.java
package com.blind75.strings;

import java.util.HashMap;
import java.util.Map;

public class LongestSubString {

  public int lengthOfLongestSubstring(String s) {
    var startIndex = 0;
    var maxLength = 0;
    Map<Character, Integer> letterIndexes = new HashMap<>();

    for (var currentIndex = 0; currentIndex < s.length(); currentIndex++) {
      var letter = s.charAt(currentIndex);
      if (letterIndexes.containsKey(letter)) {
        maxLength = Math.max(maxLength, currentIndex - startIndex);
        startIndex = Math.max(startIndex, letterIndexes.get(letter) + 1);
      }
      letterIndexes.put(letter, currentIndex);
    }
    return maxLength;
  }

}
